### ISSUE TITLE
New site doesn't load on Mobile Safari

### ISSUE THREAD

**Issue by jwoertink** [crystal-lang/crystal-website#758]
Love the new design :raised_hands:  but I'm having trouble getting it to load on mobile. I'm using iOS 17.4.1 on iPhone 13.

This is after clearing cache, and I've also tried opening up in private browsing.

![IMG_2582](https://github.com/crystal-lang/crystal-website/assets/2391/565e5592-dbfb-4b00-87f8-830d90210dc6)

After a few attempts, I was able to get the logo to load as well, but still nothing else.

**Comment by straight-shoota** [comment]
That's odd. We had a test with iOS 17.3.1 where everything was fine. 🤔

**Comment by jwoertink** [comment]
I tried connecting to the safari debugger to see if I could get anymore info, but the network panel looked fine, and there weren't any issues in the console... Though, I'm not sure I trust the debugger since updating my mac :confused:  I have issues with it on my own site. I'll see if I can get any other info on this.

**Comment by straight-shoota** [comment]
I did another test with iPad Pro 12.9", 2nd Generation iPadOS 17.2 (21C62) and everything looks fine there.

**Comment by jwoertink** [comment]
Yeah, it's still doing it for me. One weird thing I noticed is the site actually loads and I see all of the elements for a split second, then they all go away except for 1 or 2 parts. Then if I try to reload the page, it just never loads. No plugins or extensions, and it happens using Wifi and 5G networks. I can do a screen recording and send it if you want. Not sure if github will let me submit a video in a comment. If you'd like, let me know where I can send a screen recording.

**Comment by jwoertink** [comment]
Just to make sure it's not just me, it looks like @russ gets the issue too

![image](https://github.com/crystal-lang/crystal-website/assets/2391/71f37abb-ca52-4a73-b31b-0ad50ef2630d)

However, the other people on my team seem to load the site fine. Clearing cache on the entire browser and private browse tab still doesn't fix it.

**Comment by russ** [comment]
Just to note, the site does load on mobile, but just content above the fold. Scrolling down you see everything just blank after Johannes Müller. iPhone 11, iOS 17.3.1

**Comment by straight-shoota** [comment]
We received another report here: https://fosstodon.org/@fnordfish@ruby.social/112205967637696757

**Comment by fnordfish** [comment]
iPhone 11 Pro, iOS 17.4.1

https://ruby.social/@fnordfish/112205967220293581

**Comment by jwoertink** [comment]
> reload seems to be blocked.

Ok, so it's not just me... That's made it very difficult to try and debug because when I open the safari dev inspector, then refresh, it never loads :joy:

**Comment by beta-ziliani** [comment]
As a workaround, given that the issue doesn't show up in iOS Simulator, I suggest to detect the user agent and replace the page with a banner stating that it doesn't work on that browser 🤷 
Apple, I hate you.

**Comment by jwoertink** [comment]
If anyone wants to do some live debugging, maybe on a discord video chat or whatever, let me know. I can hit the site while y'all watch the logs or something.

**Comment by straight-shoota** [comment]
Let's try to figure out the specific cause for this mishap. We can disable parts of the website and see if that renders.

A binary search through `_sass/main.css` and `js/bundle.js` with removing parts of those files could be a good pointer.

@jwoertink If you're willing to help, would you try running the website locally (instructions are in the README) while making some modifications? I could also prepare a couple of preview branches to test, but that's going to slow down iterating.

**Comment by jwoertink** [comment]
Cloned the site and pulled it up locally on my phone with ngrok. Good news is I'm able to recreate the issue :tada:

Removing this section seems to fix it

https://github.com/crystal-lang/crystal-website/blob/e72f1e1aa51fc9566587016aa7367e146c13c652/_sass/pages/_home.scss#L37-L53

or I guess more accurately, removing these two lines

```
background-size: clamp(30vw, max(80rem, 50vw), 250vw) auto;
background-position: clamp(-15vw, -40rem + 50vw, 0px) top;
```

**Comment by straight-shoota** [comment]
Thanks for digging into this, Jeremey! ❤️ 

This is quite interesting. Maybe Safari has an issue with rendering this big background image with lots of thin lines and transparency... It causes issues as soon as the SVG is loaded.

**Comment by beta-ziliani** [comment]
#780 still doesn't fix it on iPhone 11 Pro Max with iOS 17.4.1

**Comment by straight-shoota** [comment]
Hm, I think it's unlikely. But could you try the master branch locally again? If the fix works there, we know we need to look at caches or any other issue related to network loading.

**Comment by jwoertink** [comment]
Ok, so I pulled down master, booted it locally, threw it up on ngrok, and got the same deal... However, after removing that entire `&::before` section and a couple of refreshes, I get this

![IMG_2732](https://github.com/crystal-lang/crystal-website/assets/2391/92435eb9-5852-439f-967b-6f9acc428803)

This is what happened last time too though. After I updated the sass, I did have to refresh like 5 times before it worked. I assumed it was cache, but maybe it's actually some race condition with the sass?

Also a side note, but I had to disable livereload. The site wouldn't come up for me at all through ngrok with it enabled. I'm sure that's unrelated, but still wanted to note.

**Comment by straight-shoota** [comment]
Hm, that's too bad 😢

I guess we'll have to remove the entire pattern background then.
Or figure out how to fix this for real 🤷 

> I assumed it was cache, but maybe it's actually some race condition with the sass?

If it is a race condition, it must be in Safari.
